server/consumer.go

```diff
@@ -3332,6 +3332,26 @@ func (o *consumer) processNextMsgRequest(reply string, msg []byte) {
 		return
 	}
 
+	if len(consumerMetaData) > 0 && len(o.cfg.Metadata) > 0 {
+		var matching = true
+		for k, v := range consumerMetaData {
+			configValue, ok := o.cfg.Metadata[k]
+			if ok {
+				if configValue != v {
+					matching = false
+					break
+				}
+			} else {
+				matching = false
+				break
+			}
+		}
+		if !matching {
+			sendErr(409, "Request's medata does not match the consumer's")
```

Comment:
This should to be the same error code.

I would really like us to have separate code's at least for errors that have a very distinct way to react to them in the client.

Also this should be a const, not a literal.

Reply:
I just did the same as what the existing other 409 errors do (e.g. line 3308) 🤷‍♂️

Reply:
Typos and incorrect `'`s here

Reply:
This should not be 409. 
Client libraries should not compare strings to determine what is happening and what to do.

Reply:
We use http status codes here, and 4xx are client errors, with 409 being conflict which seems correct.